File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

unsigned int score(coordinate_t loc){
    float distance = hypot(loc.a, loc.b);
    //Never use direct equality on floats SMH!!
    return distance < 1.01 ? 10 : distance < 5.01 ? 5 : distance < 10.01 ? 1 : 0; 
}

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
    | hypot <= 1.0 = 10
    | hypot <= 5.0 = 5
    | hypot <= 10.0 = 1
    | otherwise = 0
    where hypot = sqrt(x * x + y * y)